### ISSUE TITLE
Translations shoud be used instead of string literals

### ISSUE THREAD

**Issue by NoNoNo** [getgrav/grav-plugin-login#300]
In `pages` there are multiple untranslated texts, e.g.

https://github.com/getgrav/grav-plugin-login/blob/388f1adb594bbdf706cb892b6ecf15311aae1ed1/pages/login.md?plain=1#L2

https://github.com/getgrav/grav-plugin-login/blob/388f1adb594bbdf706cb892b6ecf15311aae1ed1/pages/login.md?plain=1#L42

https://github.com/getgrav/grav-plugin-login/blob/388f1adb594bbdf706cb892b6ecf15311aae1ed1/pages/forgot.md?plain=1#L2

https://github.com/getgrav/grav-plugin-login/blob/388f1adb594bbdf706cb892b6ecf15311aae1ed1/pages/forgot.md?plain=1#L20-L22

https://github.com/getgrav/grav-plugin-login/blob/388f1adb594bbdf706cb892b6ecf15311aae1ed1/pages/profile.md?plain=1#L2

https://github.com/getgrav/grav-plugin-login/blob/388f1adb594bbdf706cb892b6ecf15311aae1ed1/pages/profile.md?plain=1#L81

https://github.com/getgrav/grav-plugin-login/blob/388f1adb594bbdf706cb892b6ecf15311aae1ed1/pages/unauthorized.md?plain=1#L2

https://github.com/getgrav/grav-plugin-login/blob/388f1adb594bbdf706cb892b6ecf15311aae1ed1/pages/unauthorized.md?plain=1#L7

https://github.com/getgrav/grav-plugin-login/blob/388f1adb594bbdf706cb892b6ecf15311aae1ed1/pages/reset.md?plain=1#L2

https://github.com/getgrav/grav-plugin-login/blob/388f1adb594bbdf706cb892b6ecf15311aae1ed1/pages/reset.md?plain=1#L33-L35

https://github.com/getgrav/grav-plugin-login/blob/388f1adb594bbdf706cb892b6ecf15311aae1ed1/pages/register.md?plain=1#L2

https://github.com/getgrav/grav-plugin-login/blob/388f1adb594bbdf706cb892b6ecf15311aae1ed1/pages/register.md?plain=1#L58-L60

---

And, en passant, there is a jump from H1 to H3 in this file:

https://github.com/getgrav/grav-plugin-login/blob/388f1adb594bbdf706cb892b6ecf15311aae1ed1/pages/reset.md?plain=1#L33-L35

**Comment by complanar** [comment]
Hey, any progress on that, yet? I translated those files manually, but of course they are overwritten on every update.
Is there a workaround, such as copying that pages into a custom theme and keeping them there?

**Comment by rhukster** [comment]
@complanar Yes, there is a supported way to do this today that survives updates, and it is not a theme override.

The plugin looks for a page at its configured route in your own site first. If it finds one, it uses yours and never reads its own copy. So all you need to do is move those pages into `user/pages/` and translate them there. Nothing under `user/pages/` is ever touched by an update.

Copy each page from the plugin into a folder named after its route, keeping the original filename (the filename is what picks the template):

```
user/plugins/login/pages/login.md        ->  user/pages/login/login.md
user/plugins/login/pages/forgot.md       ->  user/pages/forgot_password/forgot.md
user/plugins/login/pages/reset.md        ->  user/pages/reset_password/reset.md
user/plugins/login/pages/register.md     ->  user/pages/user_register/register.md
user/plugins/login/pages/profile.md      ->  user/pages/user_profile/profile.md
user/plugins/login/pages/magic.md        ->  user/pages/magic_login/magic.md
user/plugins/login/pages/unauthorized.md ->  user/pages/user_unauthorized/unauthorized.md
```

Those folder names are the default routes. If you changed any route in the Login plugin settings, use your route as the folder name instead.

Three things to watch:

1. Copy the whole file, frontmatter and all, then edit the text. The form fields are defined in the frontmatter, so a page with only a title and a heading will render the login page with no username or password field.
2. Do not put a number in front of the folder name (use `login`, not `01.login`). A numeric prefix makes the page visible in your site navigation.
3. If your site is multi-language, you can drop `login.de.md` and `login.en.md` side by side in the same folder and Grav serves the right one per language. I tested this end to end and it works.

On the original request itself: @NoNoNo was right, and it has been sitting too long. The honest reason is that until recently there was no way to emit a translated string from a page body that worked reliably. Page content in Grav 2 does not run Twig on every site, so `{{ 'SOME.KEY'|t }}` would print literally for a lot of people. We hit that exact bug in 3.8.7 and had to take Twig back out of the password reset page.

That's now solved. The shortcode-core plugin ships a `[translate]` shortcode, and Login already requires shortcode-core, so the page bodies can be keyed properly without the Twig problem. That's now on `develop`: every page title and heading goes through a translation key, and the H1 to H3 heading jump on the reset page that was flagged in the original report is fixed too.

If you have already translated the pages by hand, move them into `user/pages/` as above and you won't lose them again.

**Comment by rhukster** [comment]
Fixed in 43e17dd on `develop`, shipping in 3.9.7. All seven pages now take their titles and headings from translation keys, and the reset page's heading jump is corrected. Verified on a live site: a site-level language override changes the rendered headings, so they genuinely resolve through the keys rather than just happening to match.

**Comment by complanar** [comment]
Thank you very much for fixing this and for the hint, how those pages can be customized even further, by overriding the translation strings and by copying that pages into `/user/pages`.
Maybe this information could become part of the README or a kind of documentation of the plugin. I'm just new to grav (tried it for a few weeks) and do not know about the internals enough yet to figure this out on my own.
